independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -581,6 +594,23 @@ private static int compareAlternativeBeans(BeanInfo bean1, BeanInfo bean2) {
         return priority2.compareTo(priority1);
     }
 
+    // gets bean priority or fall back to using default value of 0 for ordering purposes
+    private static Integer getDefaultBeanPriority(BeanInfo bean) {
+        Integer beanPriority = bean.getPriority();
+        if (beanPriority == null) {
+            beanPriority = 0;
+        }
+        return beanPriority;
```

Comment:
WDYT about inheriting `@Priority` from the declaring bean for `@DefaultBean` producers, similarly to alternatives?

Reply:
I'm not a fan. It's confusing.

Reply:
Well I'm a passionate hater of that myself, but if we say it works "like with `@Alternative`", it becomes double-confusing :-)

Reply:
I am definitely -1 to inherit these from declaring class.
Let me change to wording to something like "similarly to alternatives" :)

Reply:
I've removed the comparison to alternatives to avoid confusion.